tests/factories.py

```diff
@@ -22,33 +26,46 @@ def ensure_workspace(workspace_or_name):
 def create_workspace(name):
     workspace_dir = settings.WORKSPACE_DIR / name
     workspace_dir.mkdir(exist_ok=True, parents=True)
-    return Workspace(name)
+    return api.get_workspace(name)
+
+
+def write_workspace_file(workspace, path, contents=""):
+    workspace = ensure_workspace(workspace)
+    path = workspace.root() / path
+    path.parent.mkdir(parents=True, exist_ok=True)
+    path.write_text(contents)
 
 
-def create_release_request(workspace, user=default_user, **kwargs):
+def create_release_request(workspace, user=None, **kwargs):
     workspace = ensure_workspace(workspace)
+
+    # create a default user with permission on workspace
+    if user is None:
+        user = get_user(workspaces=[workspace.name])
+
     release_request = api._create_release_request(
         workspace=workspace.name, author=user.username, **kwargs
     )
     release_request.root().mkdir(parents=True, exist_ok=True)
     return release_request
 
 
-def write_workspace_file(workspace, path, contents=""):
-    workspace = ensure_workspace(workspace)
-    path = workspace.root() / path
-    path.parent.mkdir(parents=True, exist_ok=True)
-    path.write_text(contents)
+def write_request_file(request, group, path, contents="", user=None):
+    workspace = ensure_workspace(request.workspace)
+    try:
+        workspace.abspath(path)
+    except api.FileNotFound:
+        write_workspace_file(workspace, path, contents)
 
+    # create a default user with permission on workspace
+    if user is None:  # pragma: nocover
+        user = get_user(username=request.author, workspaces=[request.workspace])
 
-def write_request_file(request, path, contents=""):
-    path = request.root() / path
-    path.parent.mkdir(parents=True, exist_ok=True)
-    path.write_text(contents)
+    api.add_file_to_request(request, relpath=path, user=user, group_name=group)
 
 
 def create_filegroup(release_request, group_name, filepaths=None):
```

Comment:
Is this helper method needed anymore? I added it when we weren't using the api to create request files for tests, so they weren't automatically getting groups created

Reply:
There's a couple of other tests that still use it I think?

I will try rework them to use the write_release_request_files changes